### PR TITLE
[python] add "python_requires" keyword argument for setup()

### PR DESCRIPTION
**Pull request by PyVCEchecker**
Add `python_requires='>=3.5'` for **setup()** #5004

### CODE REVIEW

File: python-package/setup.py [hunk 1]
```diff
@@ -340,6 +340,7 @@ def run(self) -> None:
           version=version,
           description='LightGBM Python Package',
           long_description=readme,
+          python_requires='>=3.5',
```
Comment (StrikerRUS) [inline python-package/setup.py:343]: Thanks for this contribution!
But I don't think that LightGBM is compatible with Python 3.5 due to using f-strings e.g. here
https://github.com/microsoft/LightGBM/blob/15266f1d613a37fc6d59bc0462854a99eeee3b0b/python-package/setup.py#L134
that were introduced in Python 3.6, 
https://www.python.org/dev/peps/pep-0498/
```suggestion
          python_requires='>=3.6',
```

**Review by StrikerRUS — CHANGES_REQUESTED** [review]

**Comment by StrikerRUS** [comment]
Also, I'm re-targeting this PR to the `master` branch from `py310` one.

**Comment by StrikerRUS** [comment]
> Also, I'm re-targeting this PR to the `master` branch from `py310` one.

Ah, commits from `py310` were included into this PR by the re-targeting.
Also, CLA bot wasn't triggered for some reason.

@PyVCEchecker Could you please re-create a new PR targeting LightGBM's `master` branch and including all recent commits from it?

**Comment by PyVCEchecker** [comment]
You are right. LightGBM is not compatible with Python 3.5. I have re-create a new PR here #5006

**Pull request closed without merging**